bot/cogs/moderation/silence.py

```diff
@@ -57,23 +70,30 @@ async def _notifier(self) -> None:
 class Silence(commands.Cog):
     """Commands for stopping channel messages for `verified` role in a channel."""
 
+    # Maps muted channel IDs to their previous overwrites for send_message and add_reactions.
+    # Overwrites are stored as JSON.
+    previous_overwrites = RedisCache()
+
+    # Maps muted channel IDs to POSIX timestamps of when they'll be unsilenced.
+    # A timestamp equal to -1 means it's indefinite.
+    unsilence_timestamps = RedisCache()
+
     def __init__(self, bot: Bot):
         self.bot = bot
         self.scheduler = Scheduler(self.__class__.__name__)
-        self.muted_channels = set()
 
-        self._get_instance_vars_task = self.bot.loop.create_task(self._get_instance_vars())
-        self._get_instance_vars_event = asyncio.Event()
+        self._init_task = self.bot.loop.create_task(self._init_cog())
 
-    async def _get_instance_vars(self) -> None:
-        """Get instance variables after they're available to get from the guild."""
+    async def _init_cog(self) -> None:
+        """Set instance attributes once the guild is available and reschedule unsilences."""
```

Comment:
How would you feel about naming this `_async_init` or similar? I feel like that reflects the purpose better.

Reply:
Sure. See f218c7b